lib/_stream_readable.js

```diff
@@ -170,6 +170,10 @@ function ReadableState(options, stream, isDuplex) {
   // If true, a maybeReadMore has been scheduled.
   this.readingMore = false;
 
+  // Flags to update highwatermark and objectMode
+  this.updateHighwaterMark = null;
+  this.updateObjectMode = null;
```

Comment:
Why did you add different booleans for these? I think we could override the previous one.

Reply:
when the method is called I don't set change the HighwaterMark right away as it is in middle of _read or _write method.  I set the boolean which gets check on the next read or write and update it accordingly.

Reply:
I also agree with using setter instead of new method, just wasn't sure.  Will update that.